src/plugins/undoRedo/undoRedo.js

```diff
@@ -315,9 +315,15 @@ UndoRedo.RemoveRowAction = function(index, data) {
 inherit(UndoRedo.RemoveRowAction, UndoRedo.Action);
 
 UndoRedo.RemoveRowAction.prototype.undo = function(instance, undoneCallback) {
-  instance.alter('insert_row', this.index, this.data.length, 'UndoRedo.undo');
   instance.addHookOnce('afterRender', undoneCallback);
-  instance.populateFromArray(this.index, 0, this.data, void 0, void 0, 'UndoRedo.undo');
+  instance.alter('insert_row', this.index, this.data.length, 'undo');
+
+  var source = instance.getSourceData();
```

Comment:
It's a breaking change. Do you think it’s a better option, to change data silently (without calling any hooks)?

Reply:
@lewisjb please help us to pull this PR. It has been a very long time since our last conversation.

Reply:
I was waiting for an update on https://github.com/handsontable/handsontable/pull/3998#discussion_r190837441

Reply:
Sorry, I haven't seen that comment. I've asked @swistach to send his reply as soon as possible

Reply:
I think it's probably better to avoid a breaking change for this